### PR TITLE
[13.0.x ]ISPN-13848 Integration test - Protobuf indexed entity

### PR DESCRIPTION
**Pull request by diegolovison**
https://issues.redhat.com/browse/ISPN-13848

Backport: https://github.com/infinispan/infinispan/pull/10069/

### CODE REVIEW

**Review by karesti — CHANGES_REQUESTED** [review]
review reported in the main branch PR

File: integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java [hunk 1]
```diff
@@ -24,15 +29,19 @@ public class HotRodQueryIT extends AbstractHotRodQueryIT {
...
       war.addClass(Person.class);
       war.addClass(ITestUtils.class);
+      war.addClass(Book.class);
+      war.addPackage(RemoteQueryInitializer.class.getPackage().getName());
+      war.add(new StringAsset(getProtoContent()), "WEB-INF/classes/proto/book.proto");
       addLibrary(war, "org.infinispan:infinispan-query-dsl");
       addLibrary(war, "org.infinispan:infinispan-remote-query-client");
       addLibrary(war, "org.infinispan:infinispan-client-hotrod");
+      addLibrary(war, "commons-io:commons-io");
+      addLibrary(war, "org.infinispan:infinispan-core");
```
Comment (pruivo) [inline integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java:44]: why do you need core? and common-io?

Comment (pruivo) [reply integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java:44]: and if you remove the reflection in the test and do `RemoteQueryInitializerImpl impl = new RemoteQueryInitializerImpl();` you can just do `remoteCacheManager.getCache(protoBufCacheName).put(impl.getProtoFileName(), impl.getProtoFile());`

Comment (pruivo) [reply integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java:44]:
```diff
diff --git a/integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/remote/AbstractHotRodQueryIT.java b/integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/remote/AbstractHotRodQueryIT.java
index 2e12d26aae..94e6036e81 100644
--- a/integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/remote/AbstractHotRodQueryIT.java
+++ b/integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/remote/AbstractHotRodQueryIT.java
@@ -6,13 +6,8 @@
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
-import java.io.IOException;
-import java.io.InputStream;
-import java.net.URISyntaxException;
-import java.nio.charset.StandardCharsets;
 import java.util.List;
 
-import org.apache.commons.io.IOUtils;
 import org.infinispan.client.hotrod.RemoteCache;
 import org.infinispan.client.hotrod.RemoteCacheManager;
 import org.infinispan.client.hotrod.Search;
@@ -20,14 +15,13 @@
 import org.infinispan.client.hotrod.marshall.MarshallerUtil;
 import org.infinispan.commons.marshall.ProtoStreamMarshaller;
 import org.infinispan.protostream.SerializationContext;
-import org.infinispan.protostream.SerializationContextInitializer;
 import org.infinispan.protostream.annotations.ProtoSchemaBuilder;
 import org.infinispan.query.dsl.Query;
 import org.infinispan.query.dsl.QueryFactory;
 import org.infinispan.query.remote.client.ProtobufMetadataManagerConstants;
 import org.infinispan.test.integration.data.Book;
 import org.infinispan.test.integration.data.Person;
-import org.infinispan.test.integration.remote.proto.RemoteQueryInitializer;
+import org.infinispan.test.integration.remote.proto.RemoteQueryInitializerImpl;
 import org.junit.Assert;
 import org.junit.Test;
 
@@ -36,23 +30,16 @@
  */
 public abstract class AbstractHotRodQueryIT {
 
-   public static String getProtoContent() throws IOException {
-      try (InputStream in = AbstractHotRodQueryIT.class.getClassLoader().getResourceAsStream("proto/book.proto")) {
-         return IOUtils.toString(in, StandardCharsets.UTF_8.name());
-      }
-   }
-
    @Test
-   public void testIndexed() throws ClassNotFoundException, InstantiationException, IllegalAccessException, URISyntaxException, IOException {
-      SerializationContextInitializer serializationContextInitializerImpl =
-            (SerializationContextInitializer) Class.forName(RemoteQueryInitializer.class.getName() + "Impl").newInstance();
+   public void testIndexed() {
+      RemoteQueryInitializerImpl impl = new RemoteQueryInitializerImpl();
       ConfigurationBuilder clientBuilder = new ConfigurationBuilder()
          .addServer().host("127.0.0.1")
-         .addContextInitializers(serializationContextInitializerImpl);
+         .addContextInitializers(impl);
       try (RemoteCacheManager remoteCacheManager = new RemoteCacheManager(clientBuilder.build())) {
          // Grab the generated protobuf schema and registers in the server.
          String protoBufCacheName = ProtobufMetadataManagerConstants.PROTOBUF_METADATA_CACHE_NAME;
-         remoteCacheManager.getCache(protoBufCacheName).put("book.proto", getProtoContent());
+         remoteCacheManager.getCache(protoBufCacheName).put(impl.getProtoFileName(), impl.getProtoFile());
 
          String entity = "book_sample.Book";
 
diff --git a/integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java b/integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java
index 23ee9952e7..e33382ecab 100644
--- a/integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java
+++ b/integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java
@@ -36,11 +36,10 @@ public static Archive<?> deployment() throws IOException {
       war.addClass(ITestUtils.class);
       war.addClass(Book.class);
       war.addPackage(RemoteQueryInitializer.class.getPackage().getName());
-      war.add(new StringAsset(getProtoContent()), "WEB-INF/classes/proto/book.proto");
+      war.addAsResource("proto/book.proto");
       addLibrary(war, "org.infinispan:infinispan-query-dsl");
       addLibrary(war, "org.infinispan:infinispan-remote-query-client");
       addLibrary(war, "org.infinispan:infinispan-client-hotrod");
-      addLibrary(war, "commons-io:commons-io");
       addLibrary(war, "org.infinispan:infinispan-core");
       return war;
    }
diff --git a/integrationtests/server-integration/wildfly-modules/src/test/java/org/infinispan/test/integration/as/client/HotRodQueryIT.java b/integrationtests/server-integration/wildfly-modules/src/test/java/org/infinispan/test/integration/as/client/HotRodQueryIT.java
index 833496ea2d..5e89a23d4e 100644
--- a/integrationtests/server-integration/wildfly-modules/src/test/java/org/infinispan/test/integration/as/client/HotRodQueryIT.java
+++ b/integrationtests/server-integration/wildfly-modules/src/test/java/org/infinispan/test/integration/as/client/HotRodQueryIT.java
@@ -33,7 +33,7 @@ public static Archive<?> deployment() throws IOException {
             .create(WebArchive.class, "remote-query.war")
             .addClasses(HotRodQueryIT.class, AbstractHotRodQueryIT.class, Person.class, Book.class)
             .addPackage(RemoteQueryInitializer.class.getPackage().getName())
-            .add(new StringAsset(getProtoContent()), "WEB-INF/classes/proto/book.proto")
+            .addAsResource("proto/book.proto")
             .add(manifest(), "META-INF/MANIFEST.MF");
    }

```

Comment (diegolovison) [reply integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java:44]: **remove the reflection**

Changes applied in https://github.com/infinispan/infinispan/pull/10069

```
Index: integrationtests/server-integration/wildfly-modules/src/test/java/org/infinispan/test/integration/as/client/HotRodQueryIT.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/integrationtests/server-integration/wildfly-modules/src/test/java/org/infinispan/test/integration/as/client/HotRodQueryIT.java b/integrationtests/server-integration/wildfly-modules/src/test/java/org/infinispan/test/integration/as/client/HotRodQueryIT.java
--- a/integrationtests/server-integration/wildfly-modules/src/test/java/org/infinispan/test/integration/as/client/HotRodQueryIT.java	(revision df3fb86d5db50e80814f7c2f75af07ef98cf80a0)
+++ b/integrationtests/server-integration/wildfly-modules/src/test/java/org/infinispan/test/integration/as/client/HotRodQueryIT.java	(date 1650991088655)
@@ -33,7 +33,7 @@
             .create(WebArchive.class, "remote-query.war")
             .addClasses(HotRodQueryIT.class, AbstractHotRodQueryIT.class, Person.class, Book.class)
             .addPackage(BookQuerySchema.class.getPackage().getName())
-            .add(new StringAsset(getProtoContent()), "WEB-INF/classes/proto/book.proto")
+            .addAsResource("proto/book.proto")
             .add(manifest(), "META-INF/MANIFEST.MF");
    }
 
Index: integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/remote/AbstractHotRodQueryIT.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/remote/AbstractHotRodQueryIT.java b/integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/remote/AbstractHotRodQueryIT.java
--- a/integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/remote/AbstractHotRodQueryIT.java	(revision df3fb86d5db50e80814f7c2f75af07ef98cf80a0)
+++ b/integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/remote/AbstractHotRodQueryIT.java	(date 1650990775883)
@@ -6,12 +6,8 @@
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
-import java.io.IOException;
-import java.io.InputStream;
-import java.nio.charset.StandardCharsets;
 import java.util.List;
 
-import org.apache.commons.io.IOUtils;
 import org.infinispan.client.hotrod.RemoteCache;
 import org.infinispan.client.hotrod.RemoteCacheManager;
 import org.infinispan.client.hotrod.Search;
@@ -19,7 +15,6 @@
 import org.infinispan.client.hotrod.marshall.MarshallerUtil;
 import org.infinispan.commons.marshall.ProtoStreamMarshaller;
 import org.infinispan.protostream.SerializationContext;
-import org.infinispan.protostream.SerializationContextInitializer;
 import org.infinispan.protostream.annotations.ProtoSchemaBuilder;
 import org.infinispan.query.dsl.Query;
 import org.infinispan.query.dsl.QueryFactory;
@@ -27,6 +22,7 @@
 import org.infinispan.test.integration.data.Book;
 import org.infinispan.test.integration.data.Person;
 import org.infinispan.test.integration.remote.proto.BookQuerySchema;
+import org.infinispan.test.integration.remote.proto.BookQuerySchemaImpl;
 import org.junit.Assert;
 import org.junit.Test;
 
@@ -36,17 +32,16 @@
 public abstract class AbstractHotRodQueryIT {
 
    private static final String HOST = "127.0.0.1";
-   private static final String REMOTE_QUERY_INITIALIAZER_IMP = BookQuerySchema.class.getName() + "Impl";
 
    @Test
    public void testIndexed() throws Exception {
-      SerializationContextInitializer serializationContextInitializerImpl =
-            (SerializationContextInitializer) Class.forName(REMOTE_QUERY_INITIALIAZER_IMP).newInstance();
-      ConfigurationBuilder config = localServerConfiguration()
-         .addContextInitializers(serializationContextInitializerImpl);
+      BookQuerySchema schema = new BookQuerySchemaImpl();
+      ConfigurationBuilder config = localServerConfiguration();
+      config.addContextInitializer(schema);
       try (RemoteCacheManager remoteCacheManager = new RemoteCacheManager(config.build())) {
 
-         registerSchema(remoteCacheManager, "book.proto", getProtoContent());
+         // register schema
+         registerSchema(remoteCacheManager, schema.getProtoFileName(), schema.getProtoFile());
 
          String entity = "book_sample.Book";
 
@@ -77,13 +72,13 @@
       ConfigurationBuilder config = localServerConfiguration();
       config.marshaller(new ProtoStreamMarshaller());
       try (RemoteCacheManager rcm = new RemoteCacheManager(config.build())) {
+         // register schema
          SerializationContext serializationContext = MarshallerUtil.getSerializationContext(rcm);
          String protoKey = "test.proto";
          ProtoSchemaBuilder protoSchemaBuilder = new ProtoSchemaBuilder();
          String protoFile = protoSchemaBuilder.fileName(protoKey)
                .addClass(Person.class)
                .build(serializationContext);
-
          registerSchema(rcm, protoKey, protoFile);
 
          RemoteCache<String, Person> cache = rcm.getCache();
@@ -114,13 +109,13 @@
       ConfigurationBuilder config = localServerConfiguration();
       config.marshaller(new ProtoStreamMarshaller());
       try (RemoteCacheManager rcm = new RemoteCacheManager(config.build())) {
+         // register schema
          SerializationContext serializationContext = MarshallerUtil.getSerializationContext(rcm);
          String protoKey = "test.proto";
          ProtoSchemaBuilder protoSchemaBuilder = new ProtoSchemaBuilder();
          String protoFile = protoSchemaBuilder.fileName(protoKey)
                .addClass(Person.class)
                .build(serializationContext);
-
          registerSchema(rcm, protoKey, protoFile);
 
          RemoteCache<String, Person> cache = rcm.getCache();
@@ -146,10 +141,4 @@
       metadataCache.put(key, protoFile);
       assertFalse(metadataCache.containsKey(ProtobufMetadataManagerConstants.ERRORS_KEY_SUFFIX));
    }
-
-   public static String getProtoContent() throws IOException {
-      try (InputStream in = AbstractHotRodQueryIT.class.getClassLoader().getResourceAsStream("proto/book.proto")) {
-         return IOUtils.toString(in, StandardCharsets.UTF_8.name());
-      }
-   }
 }
Index: integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java b/integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java
--- a/integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java	(revision df3fb86d5db50e80814f7c2f75af07ef98cf80a0)
+++ b/integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java	(date 1650990950315)
@@ -14,7 +14,6 @@
 import org.jboss.arquillian.container.test.api.TargetsContainer;
 import org.jboss.arquillian.junit.Arquillian;
 import org.jboss.shrinkwrap.api.Archive;
-import org.jboss.shrinkwrap.api.asset.StringAsset;
 import org.jboss.shrinkwrap.api.spec.WebArchive;
 import org.junit.runner.RunWith;
 
@@ -36,11 +35,10 @@
       war.addClass(ITestUtils.class);
       war.addClass(Book.class);
       war.addPackage(BookQuerySchema.class.getPackage().getName());
-      war.add(new StringAsset(getProtoContent()), "WEB-INF/classes/proto/book.proto");
+      war.addAsResource("proto/book.proto");
       addLibrary(war, "org.infinispan:infinispan-query-dsl");
       addLibrary(war, "org.infinispan:infinispan-remote-query-client");
       addLibrary(war, "org.infinispan:infinispan-client-hotrod");
-      addLibrary(war, "commons-io:commons-io");
       addLibrary(war, "org.infinispan:infinispan-core");
       return war;
    }

```

Comment (diegolovison) [reply integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java:44]: **config as xml**

Changes applied in https://github.com/infinispan/infinispan/pull/10069

```
Index: integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java b/integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java
--- a/integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java	(revision 6b4a7a44dafd80309c46423f1dc420f5af984b20)
+++ b/integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/remote/HotRodQueryIT.java	(date 1650991752039)
@@ -39,7 +39,6 @@
       addLibrary(war, "org.infinispan:infinispan-query-dsl");
       addLibrary(war, "org.infinispan:infinispan-remote-query-client");
       addLibrary(war, "org.infinispan:infinispan-client-hotrod");
-      addLibrary(war, "org.infinispan:infinispan-core");
       return war;
    }
 }
Index: integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/remote/AbstractHotRodQueryIT.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/remote/AbstractHotRodQueryIT.java b/integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/remote/AbstractHotRodQueryIT.java
--- a/integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/remote/AbstractHotRodQueryIT.java	(revision 6b4a7a44dafd80309c46423f1dc420f5af984b20)
+++ b/integrationtests/server-integration/server-integration-commons/src/test/java/org/infinispan/test/integration/remote/AbstractHotRodQueryIT.java	(date 1650991699202)
@@ -1,6 +1,5 @@
 package org.infinispan.test.integration.remote;
 
-import static org.infinispan.configuration.cache.IndexStorage.FILESYSTEM;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
@@ -13,6 +12,7 @@
 import org.infinispan.client.hotrod.Search;
 import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
 import org.infinispan.client.hotrod.marshall.MarshallerUtil;
+import org.infinispan.commons.configuration.StringConfiguration;
 import org.infinispan.commons.marshall.ProtoStreamMarshaller;
 import org.infinispan.protostream.SerializationContext;
 import org.infinispan.protostream.annotations.ProtoSchemaBuilder;
@@ -43,15 +43,16 @@
          // register schema
          registerSchema(remoteCacheManager, schema.getProtoFileName(), schema.getProtoFile());
 
-         String entity = "book_sample.Book";
-
-         org.infinispan.configuration.cache.ConfigurationBuilder cacheConfig =
-               new org.infinispan.configuration.cache.ConfigurationBuilder();
-         cacheConfig.indexing().enable().storage(FILESYSTEM).path("${java.io.tmpdir}/ispn-index")
-               .addIndexedEntity(entity);
-
+         String xmlConfig = "" +
+               "<distributed-cache name=\"books\">\n" +
+               "  <indexing path=\"${java.io.tmpdir}/index\">\n" +
+               "    <indexed-entities>\n" +
+               "      <indexed-entity>" + Book.class.getName() + "</indexed-entity>\n" +
+               "    </indexed-entities>\n" +
+               "  </indexing>\n" +
+               "</distributed-cache>";
          RemoteCache<Object, Object> remoteCache =
-               remoteCacheManager.administration().getOrCreateCache("books", cacheConfig.build());
+               remoteCacheManager.administration().getOrCreateCache("books", new StringConfiguration(xmlConfig));
 
          // Add some Books
          Book book1 = new Book("Infinispan in Action", "Learn Infinispan with using it", 2015);
@@ -61,7 +62,7 @@
          remoteCache.put(2, book2);
 
          QueryFactory queryFactory = Search.getQueryFactory(remoteCache);
-         Query<Book> query = queryFactory.create(String.format("FROM %s WHERE title:'java'", entity));
+         Query<Book> query = queryFactory.create("FROM book_sample.Book WHERE title:'java'");
          List<Book> list = query.execute().list();
          assertEquals(1, list.size());
       }
```

**Comment by diegolovison** [comment]
Backported changes from https://github.com/infinispan/infinispan/pull/10069

**Pull request closed without merging**